realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -73,8 +73,9 @@
 
     static volatile Context applicationContext;
 
+    // FIXME: document this
     // Thread pool for all async operations (Query & transaction)
-    static final RealmThreadPoolExecutor asyncTaskExecutor = RealmThreadPoolExecutor.newDefaultExecutor();
+    public static final RealmThreadPoolExecutor asyncTaskExecutor = RealmThreadPoolExecutor.newDefaultExecutor();
```

Comment:
Should we have two thread pools? One for writes and one for Reads? The pool for writes just need 1 thread since you cannot parallelize these anyway.

Reply:
I've ended up adding one public thread pool with one thread which is used only for `executeTransactionAwait`. I don't think it's needed to add an additional one for reads as flows need to be collected on the same thread the query was performed.